internal/storage/bucket_handle.go

```diff
@@ -41,8 +41,12 @@ func (bh *bucketHandle) NewReader(
 	ctx context.Context,
 	req *gcs.ReadObjectRequest) (rc io.ReadCloser, err error) {
 	// Initialising the starting offset and the length to be read by the reader.
-	start := int64((*req.Range).Start)
-	end := int64((*req.Range).Limit)
+	start := int64(0)
+	end := int64(-1)
```

Comment:
setting end to -1 and computing length from it is confusing. can you directly set the length here and update it in the if block below like this:

length: -1
if(rangle !=nil) {
end = int64(req.Range.limit)
length = end-start
}

Reply:
fixed